From patchwork Tue Oct  9 02:41:15 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10631785
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 33EFB14DB
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:41:56 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 289D829B49
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:41:56 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 1CCCB29B53; Tue,  9 Oct 2018 02:41:56 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id CA3B529B49
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:41:55 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726717AbeJIJ4d (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 9 Oct 2018 05:56:33 -0400
Received: from mail-pl1-f196.google.com ([209.85.214.196]:36142 "EHLO
        mail-pl1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726692AbeJIJ4d (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 9 Oct 2018 05:56:33 -0400
Received: by mail-pl1-f196.google.com with SMTP id y11-v6so53104plt.3
        for <kvm@vger.kernel.org>; Mon, 08 Oct 2018 19:41:53 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=zSdS2HNa77GKrgMsiqKS5grJTDAwQbkISNHxM1aOUsI=;
        b=udgSOG+4R80Mh1iGxFk9VqeZHOYi7ysyyccRfe2r39ABngV8lecclMQ+pwXDCq5JyH
         cl8bk2F92Pkgv29PnRYQJxlGgjtYvpYU/R2IdLV6uEOzq7STZcdNQhcwxp3wIJQmSt3O
         LnfrFm1YBxR39njxHeNRrdiCPV8SqCETIVhi37jZH8BSQz+mu4gXnU5jqyrDjDr8jis3
         nIbwLpUsXI3Z9WcNDWVEVaQx1nMOXZIS/2uHBbeoU6YN6g3ZbksMWIFZwe66Oj8301+0
         0y9OGjfMXnPmGEpW2MCo6WMB6ctYXBfnloJCBfRQQ56ctN5BrhxfipjaHQjnWNDTqZ/z
         TuWQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=zSdS2HNa77GKrgMsiqKS5grJTDAwQbkISNHxM1aOUsI=;
        b=Z/lFw2j5PiaVyxgGjAwnld4+EoQFe5XoQcEhqHT+vS59kCCjvLjZYppsTTKGPl9sci
         ObldlV/UyM4V2ZrbGe/TZVPNocBf5bcsGddVw17n0e7camHVl9sAyEHA/Q0cwNRqOgtz
         fyVG83j4Fs1Fw1nCR9hVh0i/KTPHBTwmKH0KfA74KlmpRWmGTu1FoLaZAbJjS2qvAXTR
         uKdMzGw22up4yM1/M336Ipbk4NcUOV1J4k3byCZ2fG9tXInsTO06Jpz8RGGX9ThW90k/
         /l4NeBjYt7KnaJ3/Uazf6QnmBApXsrofAN/zp/1+QyuvR0gULpK2RH59TL+Bhtqzgofs
         10jA==
X-Gm-Message-State: ABuFfoiwsHkuphWUjUYoCyB6AyV25YARg3iHTi5Yy9G2qPN5apyRT2+M
        ohxX4aBSJA2WAV7ISBCRXU0=
X-Google-Smtp-Source: 
 ACcGV60dUCuQSCFDv0S3fSGXftbTIoIAM1GpJRbVvyboA7U3YhO/fUmfwhy5IDRucnkU8djnQI5hQw==
X-Received: by 2002:a17:902:744c:: with SMTP id
 e12-v6mr26682808plt.186.1539052913221;
        Mon, 08 Oct 2018 19:41:53 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 g5-v6sm27997183pfk.160.2018.10.08.19.41.52
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 08 Oct 2018 19:41:52 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: kvm@vger.kernel.org, Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH 1/2] KVM: refine the comment of function
 gfn_to_hva_memslot_prot()
Date: Tue,  9 Oct 2018 10:41:15 +0800
Message-Id: <20181009024116.48603-2-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
In-Reply-To: <20181009024116.48603-1-richard.weiyang@gmail.com>
References: <20181009024116.48603-1-richard.weiyang@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

The original comment is little hard to understand.

No functional change, just amend the comment a little.

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 virt/kvm/kvm_main.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index c8fb3a9d81fa..660e2e7d382f 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -1311,8 +1311,12 @@ unsigned long kvm_vcpu_gfn_to_hva(struct kvm_vcpu *vcpu, gfn_t gfn)
 EXPORT_SYMBOL_GPL(kvm_vcpu_gfn_to_hva);
 
 /*
- * If writable is set to false, the hva returned by this function is only
- * allowed to be read.
+ * Return the hva of a @gfn and the R/W attribute if possible.
+ *
+ * @slot: the kvm_memory_slot which contains @gfn
+ * @gfn: the gfn to be translated
+ * @writable: used to return the read/write attribute of the @slot if the hva
+ * is valid and @writable is not NULL
  */
 unsigned long gfn_to_hva_memslot_prot(struct kvm_memory_slot *slot,
 				      gfn_t gfn, bool *writable)

From patchwork Tue Oct  9 02:41:16 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10631787
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A6FF914BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:42:03 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 9C49629B45
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:42:03 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 908B929B4F; Tue,  9 Oct 2018 02:42:03 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 4320529B45
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 02:42:03 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726725AbeJIJ4l (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 9 Oct 2018 05:56:41 -0400
Received: from mail-pf1-f194.google.com ([209.85.210.194]:43317 "EHLO
        mail-pf1-f194.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726692AbeJIJ4l (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 9 Oct 2018 05:56:41 -0400
Received: by mail-pf1-f194.google.com with SMTP id p24-v6so41624pff.10
        for <kvm@vger.kernel.org>; Mon, 08 Oct 2018 19:42:00 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:in-reply-to:references;
        bh=kSULpQYSMln+fRwtH9r3dVLHhkcXAQPBnRdu1IWV77o=;
        b=hX4wHijVqBuxR1MsCQFStemVBL/pFZ/q+rXjko3KFMEQ/1w4qfpQJEVV/B+gCdNBpd
         VXCcVGbc535qDhgqaJ29UxqB8dXL9VC/eLzPSNqeAG/aVYMuGyhcdm+63xEV/aRJ+JUw
         54W3nbwPPlrSHW9jJvDAKyb00pl5cFX+l51WICeS6Dd19a+iOkCe01B9jwu3BlVl/UAL
         QGeYGpgChrBW6uUj52IgW9laAGc2r6HAOdgIvM6SgsgC9r6d+4Xg+yGEtRW/x3n3U17r
         rJwvvygd+agAHJC9/IUqjnb0wkx5OS+tXRFkRukH7fzKV61qt5ocAHIKaghKyGJXtSBG
         A0/Q==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references;
        bh=kSULpQYSMln+fRwtH9r3dVLHhkcXAQPBnRdu1IWV77o=;
        b=tTdFpZIeDhX0kAzW1GU0LVLpVIPrkaEjLDle4hk4HL1JIYi83rAOruH8EuS+MqK7wK
         6cTdlVCmbaKszDAIVQRAFUPrmR9HrXASFtuZrwIvfzJJUKY5ZMBBJLWMCmop52MkCi+q
         mV71D0E87F+30joNR0VEb4HAe7Kjbouv7JgIRYIQyAmnrz097kVnIat16b4I+/HaFJHZ
         Og7H8rX58TwfrAFFkyiAfIzVW5mrE2dJAoHTGE573tiJUOV0PMpMPziICot3vetjPjvg
         4/ZrUcL1RMh7hgviuvbWvsOuqqEhFoDkSlonXy8J6cQhtI8uINcUN5ZWIaI5gjZu5rQL
         zJ2g==
X-Gm-Message-State: ABuFfoga7ZJiuC8xTMFecwKY4wu0CjJ3hVVTnipDKzyV1IHrpsA6s6GF
        GUMZZTSnNST7T22V6HLFRB4mnEPq
X-Google-Smtp-Source: 
 ACcGV61x3mMtVzNKgVqvsMyZGgP2VyCAEJpiktdP4hEp6l98T7vq0X5Yd1JkEocQb0x0UwwroWHm5g==
X-Received: by 2002:a63:ff23:: with SMTP id
 k35-v6mr23726954pgi.62.1539052920397;
        Mon, 08 Oct 2018 19:42:00 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 y185-v6sm19896863pgd.8.2018.10.08.19.41.59
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 08 Oct 2018 19:41:59 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: kvm@vger.kernel.org, Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH 2/2] KVM: refine __gfn_to_pfn_memslot() a little
Date: Tue,  9 Oct 2018 10:41:16 +0800
Message-Id: <20181009024116.48603-3-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
In-Reply-To: <20181009024116.48603-1-richard.weiyang@gmail.com>
References: <20181009024116.48603-1-richard.weiyang@gmail.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Current implementation writes *false* to @writable in three different
places. Since @writable is an indicate for caller and it will be
overwritten in hva_to_pfn(), it is save to write *false* at the beginning.

After doing so, we could

  * collapse two error hva case
  * remove the check on @writable if memslot_is_readonly()

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---
 virt/kvm/kvm_main.c | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 660e2e7d382f..4b01e19a9b81 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -1539,23 +1539,19 @@ kvm_pfn_t __gfn_to_pfn_memslot(struct kvm_memory_slot *slot, gfn_t gfn,
 {
 	unsigned long addr = __gfn_to_hva_many(slot, gfn, NULL, write_fault);
 
-	if (addr == KVM_HVA_ERR_RO_BAD) {
-		if (writable)
-			*writable = false;
-		return KVM_PFN_ERR_RO_FAULT;
-	}
+	if (writable)
+		*writable = false;
 
 	if (kvm_is_error_hva(addr)) {
-		if (writable)
-			*writable = false;
-		return KVM_PFN_NOSLOT;
+		if (addr == KVM_HVA_ERR_RO_BAD)
+			return KVM_PFN_ERR_RO_FAULT;
+		else
+			return KVM_PFN_NOSLOT;
 	}
 
 	/* Do not map writable pfn in the readonly memslot. */
-	if (writable && memslot_is_readonly(slot)) {
-		*writable = false;
+	if (memslot_is_readonly(slot))
 		writable = NULL;
-	}
 
 	return hva_to_pfn(addr, atomic, async, write_fault, writable);
 }
